src/sdam/description/topology.rs

```diff
@@ -205,7 +205,7 @@ impl TopologyDescription {
         &self,
         address: &ServerAddress,
         command: &mut Command,
-        criteria: Option<&SelectionCriteria>,
+        criteria: &SelectionCriteria,
```

Comment:
Another side effect: this method no longer receives an `Option`, so it could be slightly simplified.